lib/inbox.rb

```diff
@@ -305,17 +318,23 @@ def deltas(cursor, exclude_types=[], expanded_view=false)
       end
     end
 
-    def delta_stream(cursor, exclude_types=[], timeout=0, expanded_view=false)
+    def delta_stream(cursor, exclude_types=[], include_types=[], timeout=0, expanded_view=false)
       raise 'Please provide a block for receiving the delta objects' if !block_given?
 
-      exclude_string = ""
+      if exclude_types.any? and include_types.any?
+        raise "Cannot pass both include_types and exclude_types parameters"
+      end
 
       if exclude_types.any?
-        exclude_string = _build_exclude_types(exclude_types)
+        filter_string = _build_exclude_types(exclude_types)
+      end
+
+      if include_types.any?
+        filter_string = _build_include_types(include_types)
       end
 
       # loop and yield deltas indefinitely.
-      path = self.url_for_path("/delta/streaming?exclude_folders=false&cursor=#{cursor}#{exclude_string}")
+      path = self.url_for_path("/delta/streaming?exclude_folders=false&cursor=#{cursor}#{filter_string}")
```

Comment:
Okay then make it explicit by having an if/else block instead of two consecutive if blocks in L328-332.